### PR TITLE
docs: update explanation for compile and java sdk

### LINKED ISSUE #797: `make` command with the specified install dir
We should let users be able to specify their binary install dir in the building process. Please add this option and revise the related info in the compile & install document.

### PR DESCRIPTION
**Pull request by aceforeverd**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

close #797 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by aceforeverd** [comment]
[compilation improvement](https://trello.com/c/VcpqXsbY/3-compilation-improvement)

File: docs/cn/compile.md [hunk 1]
```diff
@@ -28,21 +28,33 @@
...
+  默认: OFF
 
 - TESTING_ENABLE：是否编译测试目标
 
-  Default: OFF
+  默认: OFF
+
+- NPROC: 并发编译数
+
+  默认: $(nproc)
```
Comment (lumianph) [inline docs/cn/compile.md:53]: what is $(nproc), a system variable?

Comment (aceforeverd) [reply docs/cn/compile.md:53]: A command line utility from gnu coreutils, it prints the number of processor unit

Comment (lumianph) [reply docs/cn/compile.md:53]: i see, LGTM

**Review by dl239 — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Comment by aceforeverd** [comment]
[compilation improvement](https://trello.com/c/VcpqXsbY/3-compilation-improvement)

**Pull request merged**